cmake/module/Maintenance.cmake

```diff
@@ -30,11 +30,7 @@ function(add_maintenance_targets)
   else()
     set(exe_format ELF)
   endif()
-  if(CMAKE_CROSSCOMPILING)
-    list(JOIN DEPENDS_C_COMPILER_WITH_LAUNCHER " " c_compiler_command)
-  else()
-    set(c_compiler_command ${CMAKE_C_COMPILER})
-  endif()
+  set(c_compiler_command "${CMAKE_C_COMPILER} ${CMAKE_C_COMPILER_ARG1}")
```

Comment:
Shouldn't this just be `CMAKE_C_COMPILER`? Or why has it now changed from that?

Reply:
Please consider the following example:
```
$ cat CMakeLists.txt 
cmake_minimum_required(VERSION 3.22)
set(CMAKE_CXX_COMPILER clang++ -stdlib=libc++ -std=c++20)
project(compiler_demo CXX)
message("CMAKE_CXX_COMPILER: ${CMAKE_CXX_COMPILER}")
message("CMAKE_CXX_COMPILER_ARG1: ${CMAKE_CXX_COMPILER_ARG1}")
$ cmake -B build
-- The CXX compiler identification is Clang 18.1.3
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMAKE_CXX_COMPILER: /usr/bin/clang++
CMAKE_CXX_COMPILER_ARG1: -stdlib=libc++ -std=c++20
-- Configuring done (0.2s)
-- Generating done (0.0s)
-- Build files have been written to: /home/hebasto/CMAKE_COMPILER_DEMO/build
```

When a language being initialized, CMake splits the compiler command into two parts, i.e., a binary and its options.